### ISSUE TITLE
MacOS issues "multiple copies of this font are installed" warning (for example for JetBrainsMono Nerd Font)

### ISSUE THREAD

**Issue by martinburger** [ryanoasis/nerd-fonts#542]
#### 🎯 Subject of the issue

I installed the JetBrainsMono Nerd Font on my macOS machine by (1) installing [JetBrainsMono.zip](https://github.com/ryanoasis/nerd-fonts/releases/download/v2.1.0/JetBrainsMono.zip) manually and (2) running `brew install font-jetbrains-mono-nerd-font`.

Please note that I removed all the font files installed by step (1) before I continued with step (2).

In both cases, I get various "multiple copies of this font are installed" warnings in Font Book.

#### 🔧 Your Setup

- JetBrainsMono Nerd Font (see above)
- iterm2 Build 3.4.1
- macOS Catalina 10.15.7 (19H2)

#### ★ Optional

<img width="1229" alt="Screen Shot 2020-11-20 at 20 34 50" src="https://user-images.githubusercontent.com/954561/99842586-65d93f80-2b70-11eb-8d56-46031845095c.png">

**Comment by jaydorsey** [comment]
I'm also seeing this. Comparing the 3 fonts that are duplicated, the only collision that seems to make sense is that the "Postscript Name" is identical for the 3 fonts I've installed.

I'm seeing the same behavior with the 3 Italic fonts in the Mono versions.

**Comment by jaydorsey** [comment]
I was able to:

- Remove the "duplicate" copies of the font, leaving regular Italic installed
  - ExtraBold Italic
  - Medium Italic
- Download FontForge and open the 2 "duplicate" copies
- Modify the Font Name (Element -> Font Info in FontForge)
  - I changed the fonts to MItalic (for Medium) and EBItalic (for Extra Bold) to differentiate the names
  - I ignored this error (on macos) but it might be nice to "shorten" the names so they comply
![image](https://user-images.githubusercontent.com/191564/106692863-4e242b80-65a3-11eb-9dc0-6c05fe8ac5c7.png)
  - I retained the Unique ID when prompted (guessing this will prevent duplicates if I reinstall again
- Regenerate the fonts with File -> Generate Fonts
  - Changed the type to True Type
  - Ignored the warning about Em-Size (chose Yes)
  - Ignored the errors warning (chose Generate)
- Installed my new fonts

So far everything looks OK but I need to do some more testing with the fonts. I think it would be better to fix the patcher, or for windows compatibility maybe choose a shorter name (JB instead of JetBrains?). I'm assuming ExtraBold and Medium got dropped to meet the 31 character naming warning.

**Comment by reportaman** [comment]
@martinburger @jaydorsey This is not just a warning! Terminal emulator applications [WezTerm](https://github.com/wez/wezterm) are not able to recognize all the NERD fonts like the regular italic version of jetbrain mono you mention. I suspect it has to do something with this...

**Comment by nerosnm** [comment]
I'm having the same problem with BlexMono. Font Book thinks the ExtraLight Italic, Italic, Light Italic, Medium Italic, Text Italic, Thin Italic, and SemiBold Italic faces are duplicates. Alacritty can recognise Italic, but it looks like none of the other faces are usable.

**Comment by docwhat** [comment]
The Blex Mono font seems to have the same problem.

```diff
--- /tmp/original-list	2021-05-14 09:50:17.000000000 -0400
+++ /tmp/after-automatically-removal-by-font-book	2021-05-14 09:49:51.000000000 -0400
@@ -1,18 +1,12 @@
 Blex Mono Bold Italic Nerd Font Complete.ttf
 Blex Mono Bold Nerd Font Complete.ttf
-Blex Mono ExtraLight Italic Nerd Font Complete.ttf
 Blex Mono ExtraLight Nerd Font Complete.ttf
 Blex Mono Italic Nerd Font Complete.ttf
-Blex Mono Light Italic Nerd Font Complete.ttf
 Blex Mono Light Nerd Font Complete.ttf
-Blex Mono Medium Italic Nerd Font Complete.ttf
 Blex Mono Medium Nerd Font Complete.ttf
 Blex Mono Nerd Font Complete.ttf
-Blex Mono SemiBold Italic Nerd Font Complete.ttf
 Blex Mono SemiBold Nerd Font Complete.ttf
-Blex Mono Text Italic Nerd Font Complete.ttf
 Blex Mono Text Nerd Font Complete.ttf
-Blex Mono Thin Italic Nerd Font Complete.ttf
 Blex Mono Thin Nerd Font Complete.ttf
```

**Comment by facuparedes** [comment]
Same issue here with JetBrains

**Comment by learn2reid** [comment]
Same issue with Jetbrains, what's the best way to resolve?

**Comment by Finii** [comment]
The problem is the Family und SubFamily naming that is too simple to handle whole sets of fonts. 

Would be fixed by #717.

**Comment by unikitty37** [comment]
Same issue with Hasklug:

![image](https://user-images.githubusercontent.com/139183/161386158-7307e244-2f65-4ef7-96c8-eaf292aa15d2.png)

![image](https://user-images.githubusercontent.com/139183/161386171-3e8897f1-f8eb-4f04-97a4-0521c7a7fdc6.png)

![image](https://user-images.githubusercontent.com/139183/161386324-ac10565d-b452-400b-aabc-638e422918f9.png)

**Comment by martinburger** [comment]
> In both cases, I get various "multiple copies of this font are installed" warnings in Font Book.

Version 2.2.2 seems to have this fixed. Thx!

**Comment by Finii** [comment]
> Version 2.2.2 seems to have this fixed. Thx!

Great to hear that! Thanks for the feedback :green_heart: Normally ppl just complain and never say when something works ;-)

**Comment by Finii** [comment]
Aforementioned fix by #717 has been pulled indirectly by #723, but it is not active for any font but `Caskaydia`.
You can utilize it when self patching with `--makegroups`. That option will end up finally in the `src/unpatched-fonts/*/config.cfg` files where appropriate.

Maybe the fix has been this PR less last minute commit: [`font-patcher: Correct Family for Mono Windows`](https://github.com/ryanoasis/nerd-fonts/commit/a6e6ee26d83ddc59fc2e9bf927b912195066988a)

Actually I'm astonished that the issue is gone :grinning:

The change with the families is planned for `3.0.0`, because it is kind of 'breaking' when names change.
Before that an update all the things version `2.3.0`.
`2.2.0` was about fixing 'simple' stuff mainly.

**Comment by sizur** [comment]
`VictorMono` has the same issue.

**Comment by noefroidevaux** [comment]
`Iosevka` seems to have the same issue

**Comment by Finii** [comment]
Is one of you able/willing to try a VictorMono / Iosevka / ? font set (i.e. parallel installation of all of them that we have in the release zip), and try if they work as expected?
I could generate the zips just for you to try, so maybe we can enable it then for v2.3.0.

Until now its only me who tested it (I assume), and then only on Linux :grimacing: But _theoretically_ it must fix it ;-)

**Comment by Finii** [comment]
Reopening this, removing the font specific reference in the title, to collect more complains.

**Comment by noefroidevaux** [comment]
> Is one of you able/willing to try a VictorMono / Iosevka / ? font set (i.e. parallel installation of all of them that we have in the release zip), and try if they work as expected? I could generate the zips just for you to try, so maybe we can enable it then for v2.3.0.
> 
> Until now its only me who tested it (I assume), and then only on Linux 😬 But _theoretically_ it must fix it ;-)

Happy to test, but not sure what I have to do 😊 You want me to test the v2.3.0-RC? Or you want to provide us a new zip?

**Comment by Finii** [comment]
> Or you want to provide us a new zip?

I will generate a new Iosevka.zip for you tomorrow. :-)

**Comment by noefroidevaux** [comment]
> > Or you want to provide us a new zip?
> 
> I will generate a new Iosevka.zip for you tomorrow. :-)

@Finii Where can I find the zip file?

**Comment by Finii** [comment]
@noefroidevaux :exploding_head: Sorry forgotten. I build it right away, hold tight...

**Comment by Finii** [comment]
@noefroidevaux You can find the newly patched Iosevka here:
https://github.com/ryanoasis/nerd-fonts/releases/download/v2.3.0-RC/Iosevka_makegroups.zip

The naming scheme changed with `--makegroups` a bit.
Please remove all patched Iosevkas before installing from the new `zip`.

**Comment by noefroidevaux** [comment]
@Finii Thank you! I removed the old font and installed the new one. And it seems all ok for me 😃👍

<img width="536" alt="image" src="https://user-images.githubusercontent.com/37739/199233732-d93ee164-a74c-45d1-b3f3-80405ff7d879.png">

<img width="174" alt="image" src="https://user-images.githubusercontent.com/37739/199233780-ec24dc39-3ffd-4859-b50f-4380aeeff6ec.png">

**Comment by Finii** [comment]
Great to hear! Thanks for testing!

Do you have this font book thingy? And it issues no warnings anymore?

**Comment by unikitty37** [comment]
No joy when I install them *all*, I'm afraid… (This is on macOS 12.6 (21G115). I didn't have any version of Iosevka installed before.)

![image](https://user-images.githubusercontent.com/139183/199501122-52cead60-0147-4223-8307-62b8ca2cd1eb.png)

Removing the NF versions fixes the issue and removes the messages for the Nerd Font versions; I'm not sure what the two versions are, as no filenames have NF in them. Is that the Windows ones? If so, are they even supposed to be installed together? (It might be a better idea to have Windows and Not-Windows folders to stop people installing both at once, if so…)

![image](https://user-images.githubusercontent.com/139183/199502794-eee723a8-06e3-48cc-806e-8729b9a31c96.png)

**Comment by noefroidevaux** [comment]
@unikitty37 Yes, I forgot to say that I only installed those that don't have "Windows Compatible" in the file name. Otherwise I also have the "NF" versions. /cc @Finii

**Comment by Finii** [comment]
Well, I would drop the `Windows Compatible` fonts completely, they are only needed for e.g. IE9 or some old MS-Word-for-Mac (!!!).

They should have different names, ... I will check that.

Thank you for testing :tada:

**Comment by noefroidevaux** [comment]
> Thank you for testing 🎉

Thank you to you!!!

**Comment by Finii** [comment]
This makes it 3 fonts that have `--makegroups`

* Cascadia Code
* Iosevka
* JetBrains Mono

Lets push out `v2.3.0` and see how it fares and activated it for the remaining fonts in `v3.0.0`

**Comment by aravezskinteeth** [comment]
I have the same issue with `VictorMono Nerd Font` with the Bold Oblique and Bold Italic Weights. No issues with the unpatched font. 

<img width="929" alt="SCR-20230320-hry" src="https://user-images.githubusercontent.com/104392087/226464130-d6087592-2881-4551-9504-35cfdb48f32f.png">

<img width="929" alt="SCR-20230320-hzx" src="https://user-images.githubusercontent.com/104392087/226464199-35a2ead1-4faf-4119-9c17-6f543ac533d5.png">

<img width="929" alt="SCR-20230320-i04" src="https://user-images.githubusercontent.com/104392087/226464240-2fca1e74-1bc9-4994-948b-beadf858264a.png">

**Comment by Finii** [comment]
Thanks for reporting.

Yes. All fonts suffer more of less that problem. This will be fixed in v3.0.0, which is due ... last week 😬

**Comment by aravezskinteeth** [comment]
hahaha ok I'll just wait patiently
